LayoutTests/ipc/message-listener-async-message-reply-id.html

```diff
@@ -0,0 +1,38 @@
+
+<!doctype html><!-- webkit-test-runner [ IPCTestingAPIEnabled=true ] -->
+<title>Test that intercepting an async message has access to the listener id.</title>
+<script src="../resources/testharness.js"></script>
+<script src="../resources/testharnessreport.js"></script>
+<body>
+<script>
+
+promise_test(async t => {
+    if (!window.IPC)
+        return;
+    const testerID = 447;
+    const processTarget = 'UI'; // Currently only UI is supported for addTesterProxy.
+    // Test starts here.
+    IPC.addTesterProxy(processTarget);
+    try {
+        let messages = [];
+        IPC.addIncomingMessageListener(processTarget, (message) => {
+            if (message.name != IPC.messages.IPCTesterProxy_AsyncMessage.name)
+                return;
+            messages.push(message);
+        });
+        IPC.sendMessage(processTarget, 0, IPC.messages.IPCTester_SendAsyncMessageToProxy.name, [
+            { type: 'uint32_t', value: 89 },
+        ]);
+        await t.step_wait(() => { return messages.length > 0 }, "Waiting until async message is intercepted");
+        assert_equals(messages.length, 1, `for ${ processTarget }`);
+        assert_equals(messages[0].arguments.length, 1, `for ${ processTarget }`)
+        assert_equals(messages[0].arguments[0].type, "uint32_t", `for ${ processTarget }`);
+        assert_equals(messages[0].arguments[0].value, 90, `for ${ processTarget }`);
+        assert_equals(typeof messages[0].listenerID, "number", `for ${ processTarget }`);
+        assert_greater_than(messages[0].listenerID, 0, `for ${ processTarget }`);
+    } finally {
+        IPC.removeTesterProxy(processTarget);
+    }
+}, "Intercepting async messages contains all the expected data.");
+</script>
+</body>
```

Comment:
Nit: Add a new line at the end?

Reply:
Done.
I assume by rule "test are source and source typically should end in newline so that sources are regular or old patch and diff programs are able to process them or both".
I'll open a bug about style rule for ending test .htmls, .css, .js with new line.